File: javascript/6_kyu/highest_scoring_word.js
// Highest Scoring Word
// https://www.codewars.com/kata/57eb8fcdf670e99d9b000272/train/javascript

function makeScorecard(){
    var alpha = "abcdefghijklmnopqrstuvwxyz";

    var point = 1;
    var points = {};

    for (var i = 0; i < alpha.length; i++) {
        points[alpha[i]] = point;
        point++;
    }

    return points;
}

function high(x) {
    var scorecard = makeScorecard();
    var words = x.split(" ");

    var scores = {};

    for (var i = 0; i < words.length; i++) {
        var count = 0;

        for (var j = 0; j < words[i].length; j++) {
            count += scorecard[words[i][j]];
        }

        scores[words[i]] = count;
    }

    var greatest = 0;

    for (var key in scores) {
        if ( scores[key] > greatest ) {
            var ans = key;
            greatest = scores[key];
        }
    }

    return ans;
}

console.log(high("man i need a taxi up to ubud"));
console.log(high("what time are we climbing up the volcano"));
console.log(high("take me to semynak"));

File: python/6_kyu/highest_scoring_word.py
# Highest Scoring Word
# https://www.codewars.com/kata/57eb8fcdf670e99d9b000272/train/python

def make_scorecard():
    alpha = 'abcdefghijklmnopqrstuvwxyz'
    point, points = 1, {}

    for i in alpha:
        points[i] = point
        point += 1
    
    return points

def high(x):
    scorecard = make_scorecard()
    contestants = x.split(" ")

    scores = {}

    for i in contestants:
        count = 0
        
        for j in i:
            count += scorecard[j]
        
        scores[i] = count
    
    return scores

    

print(high("man i need a taxi up to ubud"))
print(high("what time are we climbing up the volcano"))
print(high("take me to semynak"))